lib/logstash/inputs/jdbc.rb

```diff
@@ -201,6 +201,12 @@ module LogStash module Inputs class Jdbc < LogStash::Inputs::Base
   # this will only convert column0 that has ISO-8859-1 as an original encoding.
   config :columns_charset, :validate => :hash, :default => {}
 
+  config :use_prepared_statements, :validate => :boolean, :default => false
+
+  config :prepared_statement_name, :validate => :string, :default => SecureRandom.hex(40)
```

Comment:
could this potentially lead into a "prepared statements" memory leak on the database if the default is used and the plugin is repetitively restarted?

Reply:
You raise a good point about lifetimes and lifecycles of the Prepared Statements.
For some DB technologies, the prepared statements are associated with the connection  (session) - when the connection is closed the caches are cleansed. For others, I dunno.
Certainly, we should call `close` on the prepared statement Sequel object when the plugin stops however I can't find evidence that closing the prepared statement affects its cached info. I think we are OK using a random string in the event of a user forgetting to set the `prepared_statement_name ` setting but I'll add this hook into the plugin stop call anyway.

Reply:
or we make the name a required option?

Reply:
It can't be a validation as a standard required setting because it is mode dependent. I suppose I could introduce some conditional validation.

Reply:
ah I see... no need to block PR for this but it feels like a potential long term issue.